data/src/test/java/org/apache/iceberg/io/TestAppenderFactory.java

```diff
@@ -46,45 +53,40 @@
 import org.apache.iceberg.relocated.com.google.common.collect.Sets;
 import org.apache.iceberg.util.Pair;
 import org.apache.iceberg.util.StructLikeSet;
-import org.junit.Assert;
-import org.junit.Before;
-import org.junit.Test;
-import org.junit.runner.RunWith;
-import org.junit.runners.Parameterized;
-
-@RunWith(Parameterized.class)
-public abstract class TestAppenderFactory<T> extends TableTestBase {
-  private static final int FORMAT_V2 = 2;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.TestTemplate;
+import org.junit.jupiter.api.extension.ExtendWith;
 
-  private final FileFormat format;
-  private final boolean partitioned;
+@ExtendWith(ParameterizedTestExtension.class)
+public abstract class TestAppenderFactory<T> extends TestBase {
+  private static final int FORMAT_V2 = 2;
 
   private PartitionKey partition = null;
   private OutputFileFactory fileFactory = null;
 
-  @Parameterized.Parameters(name = "FileFormat={0}, Partitioned={1}")
-  public static Object[] parameters() {
-    return new Object[][] {
-      new Object[] {"avro", false},
-      new Object[] {"avro", true},
-      new Object[] {"orc", false},
-      new Object[] {"orc", true},
-      new Object[] {"parquet", false},
-      new Object[] {"parquet", true}
-    };
-  }
+  @Parameter protected FileFormat format;
+
+  @Parameter(index = 1)
+  private boolean partitioned;
 
-  public TestAppenderFactory(String fileFormat, boolean partitioned) {
-    super(FORMAT_V2);
-    this.format = FileFormat.fromString(fileFormat);
-    this.partitioned = partitioned;
+  @Parameters(name = "FileFormat={0}, partitioned={1}")
+  public static List<Object> parameters() {
+    return Arrays.asList(
+        new Object[] {FileFormat.AVRO, false},
+        new Object[] {FileFormat.AVRO, true},
+        new Object[] {FileFormat.ORC, false},
+        new Object[] {FileFormat.ORC, true},
+        new Object[] {FileFormat.PARQUET, false},
+        new Object[] {FileFormat.PARQUET, true});
   }
 
   @Override
-  @Before
+  @BeforeEach
   public void setupTable() throws Exception {
-    this.tableDir = temp.newFolder();
-    Assert.assertTrue(tableDir.delete()); // created by table create
+    this.tableDir = Files.createTempDirectory(temp, "junit").toFile();
+    assertThat(tableDir.delete()).isTrue(); // created by table create
+
+    this.formatVersion = FORMAT_V2;
```

Comment:
`formatVersion` is a parameterized field and needs to be set in `parameters()` instead of here